pkgs/development/interpreters/luajit/default.nix

```diff
@@ -88,8 +94,7 @@ stdenv.mkDerivation rec {
     "PREFIX=$(out)"
     "DEFAULT_CC=cc"
     "CROSS=${stdenv.cc.targetPrefix}"
-    # TODO: when pointer size differs, we would need e.g. -m32
```

Comment:
The note should be preserved, this special casing only works x86_64.

Reply:
Except that this comment is almost only just for x86 because that is not some universal [option](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/Option-Summary.html#Option-Summary). The only other applicable (also directly supported by nixpkgs) I found is PowerPC.

Nixpkgs provides, in general, only a single target compilers. That is why `gcc_multi` exists alongside `gcc`. That `gcc_multi` currently also hard depends on i686 and doesn't work on other platforms.

```console
$ nix shell nixpkgs#pkgsCross.aarch64-multiplatform.stdenv.cc -c aarch64-unknown-linux-gnu-gcc -E -march=help -xc /dev/null
...
cc1: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a
$ nix shell nixpkgs#pkgsCross.aarch64-multiplatform.stdenv_32bit.cc
error: Multilib aarch64-unknown-linux-gnu-stage-final-gcc-wrapper-9.5.0 not supported for ‘aarch64-linux’
(use '--show-trace' to show detailed location information)
```

I am going to add `TODO` at the other place instead of this one where support for other platforms should be required. The issue is that it won't be that easy for ARM for example because not every aarch64 supports for example armv7l.

The best practice here would be to use `stdenv_32bit` but that is broken as I described in https://github.com/NixOS/nixpkgs/issues/212494. I will suggest that in the `TODO`.